File: haskell/src/FizzBuzz.hs
module FizzBuzz where

fizzBuzz :: Int -> String
fizzBuzz n = addDefault $ fizz ++ buzz
  where
    fizz
      | n `mod` 3 == 0 = "Fizz"
      | otherwise = ""
    buzz
      | n `mod` 5 == 0 = "Buzz"
      | otherwise = ""
    addDefault "" = show n
    addDefault s = s

File: python/src/fizzbuzz.py
class FizzBuzz():
  def to_fizzbuzz(self, number: int):
    copy = number
    digit_sum = 0
    while copy > 0:
      digit_sum += copy % 10
      copy //= 10
    if digit_sum in [3, 6, 9]:
      return 'Fizz'
    return str(number)
    